Bottom: c01690954c065eac76e21f93b746dc4244ab8187
Top:    b54bd0e6a93844f3bfa09532a745fbe9acf98b79
Author: Igor Stoppa <igor.stoppa@huawei.com>
Date:   2018-05-26 12:41:07 +0300

pmalloc_h wip


---

diff --git a/include/linux/pmalloc.h b/include/linux/pmalloc.h
index 867eac4c2a86..cecf3c91604a 100644
--- a/include/linux/pmalloc.h
+++ b/include/linux/pmalloc.h
@@ -319,7 +319,6 @@ struct pmalloc_pool *pmalloc_create_custom_pool(size_t refill,
  */
 static inline struct pmalloc_pool *pmalloc_create_pool(uint8_t mode)
 {
-	pr_info("align order: %lu", PMALLOC_ALIGN_ORDER_DEFAULT);
 	return pmalloc_create_custom_pool(PMALLOC_REFILL_DEFAULT,
 					  PMALLOC_ALIGN_ORDER_DEFAULT,
 					  mode);
@@ -448,10 +447,8 @@ bool pmalloc_rare_write(struct pmalloc_pool *pool, const void *dst,
 	 * that are not supposed to be modifiable.
 	 */
 	area = __pool_get_area(pool, dst, n_bytes);
-	if (WARN(!area, "Destination range not in pool"))
-		return false;
-	if (WARN(!__is_area_rewritable(area),
-		 "Attempting to modify non rewritable area"))
+	if (WARN(!(area && __is_area_rewritable(area)),
+		 "Incorrect destination."))
 		return false;
 	return __raw_rare_write(dst, src, RARE_WRITE_VMALLOC_ADDR, n_bytes);
 }
